fabric-api-lookup-api-v1/src/main/java/net/fabricmc/fabric/api/lookup/v1/ApiLookupMap.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.lookup.v1;
+
+import java.util.Objects;
+import java.util.function.Supplier;
+
+import net.minecraft.util.Identifier;
+
+import net.fabricmc.fabric.impl.lookup.ApiLookupMapImpl;
+
+/**
+ * Access to generic Api lookup instances. This is is meant to be used by implementors of Api lookup registries.
+ *
+ * <p>Note: To store a lookup class with type parameters {@code <T, C>}, it is recommended to store it as {@code <?, ?>} and perform an unchecked cast on queries internally.
+ *
+ * @param <L> The type of the lookup.
+ */
+public interface ApiLookupMap<L> extends Iterable<L> {
```

Comment:
Unless im missing something I dont see these 2 interfaces needing to be exposed as API? They seem to be related to impl details?

Reply:
They are indeed implementation details, but they are provided to accelerate implementations of other types of api lookups.

See for example the third-party https://github.com/Technici4n/FastTransferLib/blob/transactions/src/main/java/dev/technici4n/fasttransferlib/impl/item/ItemKeyApiLookupRegistryImpl.java and https://github.com/Technici4n/FastTransferLib/blob/transactions/src/main/java/dev/technici4n/fasttransferlib/impl/item/ItemKeyApiLookupImpl.java for an example of how it can be used by other mods.

The plan is to add item and entity api lookups as well in later PRs, but this could still be useful in other contexts, basically any time you want to convert a game object to some type `T`. For example, someone might want to expose a `WorldApiLookup`.

Reply:
Ok that’s cool. Just don’t want to expose something for the sake of it, but this is fine.